lib/db.js

```diff
@@ -930,7 +871,7 @@ module.exports = (
   DB.prototype.consumeSigninCode = function (code) {
     log.trace({ op: 'DB.consumeSigninCode', code })
 
-    return this.pool.post(`/signinCodes/${code.toString('hex')}/consume`)
+    return this.pool.post(`/signinCodes/${Buffer.from(code, 'base64').toString('hex')}/consume`)
```

Comment:
As per my comment against `lib/routes/signin-codes.js`, this would be simpler if we just returned a hex value from the `hexSigninCode` function in that module and `code` was just passed verbatim here instead.